packages/block-library/src/navigation/index.php

```diff
@@ -123,6 +123,16 @@ function block_core_navigation_add_directives_to_submenu( $w, $block_attributes
 				$w->set_attribute( 'data-wp-bind--aria-expanded', 'selectors.core.navigation.isMenuOpen' );
 			};
 
+			// Add directives to the submenu.
+			if ( $w->next_tag(
+				array(
+					'tag_name'   => 'UL',
+					'class_name' => 'wp-block-navigation-submenu',
+				)
+			) ) {
+				$w->set_attribute( 'data-wp-on--focusin', 'actions.core.navigation.openMenuOnFocus' );
```

Comment:
The same as `focusout`, the `focusin` event is propagated to all the focusable children. This means that the action is called each time the focus is changed between the submenu elements. Not sure if it is a problem, but we are calling that action more times than needed. Should we add a check in the action or is it okay?

Reply:
I think it's ok. I think `openMenu` and `closeMenu` should be more declarative, and calling them multiple times should be ok.